lucuma-schemas/src/main/scala/lucuma/schemas/decoders/CoreModelDecoders.scala

```diff
@@ -135,4 +137,29 @@ trait CoreModelDecoders {
       q <- c.downField("q").as[Offset.Component[Axis.Q]]
     } yield Offset(p, q)
   )
+
+  // Copied from lucuma-odb for now ...
+  implicit val timeSpanDecoder: Decoder[TimeSpan] =
+    Decoder.instance { c =>
+      def from[T: Decoder](field: String, format: Format[T, TimeSpan]): Decoder.Result[TimeSpan] =
+        c.downField(field).as[T].flatMap { t =>
+          format
+            .getOption(t)
+            .toRight(DecodingFailure(s"Invalid TimeSpan $field: $t", c.history))
+        }
+
+      c.downField("microseconds")
+        .as[Long]
+        .flatMap { µs =>
+          TimeSpan.FromMicroseconds
+            .getOption(µs)
+            .toRight(DecodingFailure(s"Invalid TimeSpan microseconds: $µs", c.history))
+        }
+        .orElse(from("milliseconds", TimeSpan.FromMilliseconds))
+        .orElse(from("seconds", TimeSpan.FromSeconds))
+        .orElse(from("minutes", TimeSpan.FromMinutes))
+        .orElse(from("hours", TimeSpan.FromHours))
+        .orElse(from("iso", TimeSpan.FromString))
+        .orElse(DecodingFailure(s"Could not parse duration value", c.history).asLeft)
+    }
```

Comment:
If querying for units other than the canonical ones, can this result in rounding producing a different value than in the DB?

Reply:
Hmm, good question.  I think it boils down to whether `TimeSpan.FromMilliseconds` etc. are valid `Format[BigDecimal, TimeSpan]` instances?  If so, then the reverse get yields the canonical `BigDecimal` form and it round trips. Certainly it passes the format tests.  The issue would be with minutes and hours, where rounding in the big decimal representation has to happen in some cases.  I _think_ at the default precision though there isn't an issue even for hours.

Reply:
Supporting only canonical units here seems enough here, which is what we've done with other similar cases (`Angle`, `Wavelength`, etc.).

I was just wondering if we were introducing possible rounding issues by supporting the rest. Should we enable all units for the other cases?

Reply:
I just copied this from the `lucuma-odb` schema module where I tried to supply a general purpose decoder.  I suppose we all have a pending decision on the future of `lucuma-schemas` vs. schema module/libraries per service. Until then, I don't think we gain anything from enabling decoding in all units.